### PR TITLE
Compilation failure on nvcc path if use hipChannelFormatKind

### LINKED ISSUE #1183: [hipcc] Compilation error using hipChannelFormatKindSigned
Compilation terminates with following error when I use hipChannelFormatKindSigned:

```
cat test.cpp
#include <hip/hip_runtime.h>
int a()
{
	hipResourceDesc resDesc;
	resDesc.res.linear.desc.f = hipChannelFormatKindSigned;
return 0;
}
```
```
hipcc -c  test.cpp

test.cpp(5): error: a value of type "hipChannelFormatKind" cannot be assigned to an entity of type "cudaChannelFormatKind"

test.cpp(4): warning: variable "resDesc" was set but never used

1 error detected in the compilation of "/tmp/tmpxft_00000f80_00000000-6_test.cpp1.ii".
```
It works fine if I use cudaChannelFormatKindSigned instead.

```
hipcc --version

HIP version: 1.5.19244
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2017 NVIDIA Corporation
Built on Fri_Sep__1_21:08:03_CDT_2017
Cuda compilation tools, release 9.0, V9.0.176
```
```
git show --summary
commit 00f25833a95ff8ee934abe0cb18d8908af7aae7b (HEAD -> master, origin/master, origin/HEAD)
Merge: 96dc7489 db6571ae
```

### PR DESCRIPTION
**Pull request by SarbojitAMD**
Fixes #1183

### CODE REVIEW

File: include/hip/nvcc_detail/hip_runtime_api.h [hunk 1]
```diff
@@ -160,6 +160,7 @@ typedef CUmodule hipModule_t;
 typedef CUfunction hipFunction_t;
 typedef CUdeviceptr hipDeviceptr_t;
 typedef struct cudaArray hipArray;
+typedef struct cudaArray* hipArray_t;
```
Comment (mangupta) [inline include/hip/nvcc_detail/hip_runtime_api.h]: This change is addressed in another PR.

**Comment by gargrahul** [comment]
LGTM other than as commented by @mangupta about a change which is already merged through another PR.

**Comment by SarbojitAMD** [comment]
Please review only hipChannelFormatKind related changes, because of my incorrect push my two PRs has got merged here. 
Thanks

**Comment by SarbojitAMD** [comment]
Closing it now as I have raised fresh PR

**Pull request closed without merging**